### PR TITLE
Cleaning up code and enhancing a few IP management logs

### PR DESCRIPTION
**Pull request by GabrielBrascher**
### Description
This PR removes some unnecessary code and enhances a few log messages at IpAddressManagerImpl.

Additionally, method `com.cloud.network.NetworkModelImpl.areServicesEnabledInZone` does not use list `checkedProvider`; the respective list should either be removed or populated.

I have chosen to add provider names at `checkedProvider`, instead of removing it; however, I would appreciate any opinion/suggestion.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [X] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Review by DaanHoogland — APPROVED** [review]
code looks good and no functional changes to be seen

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/network/IpAddressManagerImpl.java [hunk 1]
```diff
@@ -824,11 +824,11 @@ public IPAddressVO doInTransaction(TransactionStatus status) throws Insufficient
                     if (vlanDbIds == null || vlanDbIds.contains(nonDedicatedVlan.getId()))
                         nonDedicatedVlanDbIds.add(nonDedicatedVlan.getId());
                 }
-                if (dedicatedVlanDbIds != null && !dedicatedVlanDbIds.isEmpty()) {
+                if (!dedicatedVlanDbIds.isEmpty()) {
                     fetchFromDedicatedRange = true;
                     sc.setParameters("vlanId", dedicatedVlanDbIds.toArray());
                     errorMessage.append(", vlanId id=" + Arrays.toString(dedicatedVlanDbIds.toArray()));
-                } else if (nonDedicatedVlanDbIds != null && !nonDedicatedVlanDbIds.isEmpty()) {
```
Comment (GabrielBrascher) [inline server/src/main/java/com/cloud/network/IpAddressManagerImpl.java:831]: Null check not needed, the list has been created at line 793.

`List<Long> nonDedicatedVlanDbIds = new ArrayList<Long>();`

Comment (DaanHoogland) [reply server/src/main/java/com/cloud/network/IpAddressManagerImpl.java:831]: yes, you could have told me before i checked it ;)

**Comment by GabrielBrascher** [comment]
A conflict has emerged, I will reserve some time to fix it.

**Comment by GabrielBrascher** [comment]
Conflicts have been fixed.
@blueorangutan package

**Comment by GabrielBrascher** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Comment by GabrielBrascher** [comment]
Thanks for the review, @sureshanaparti.
@DaanHoogland I am not sure why the Marving tests are failing. I assume that it is not related to this PR's changes.

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by sureshanaparti** [comment]
triggering the tests again to check the latest results.
@blueorangutan test

File: server/src/main/java/com/cloud/network/IpAddressManagerImpl.java [hunk 2]
```diff
@@ -1124,15 +1124,18 @@ public boolean applyIpAssociations(Network network, boolean postApplyRules, bool
...
     @DB
     @Override
     public AcquirePodIpCmdResponse allocatePodIp(String zoneId, String podId) throws ConcurrentOperationException, ResourceAllocationException {
 
         DataCenter zone = _entityMgr.findByUuid(DataCenter.class, zoneId);
         Account caller = CallContext.current().getCallingAccount();
         if (Grouping.AllocationState.Disabled == zone.getAllocationState() && !_accountMgr.isRootAdmin(caller.getId())) {
-            ResourceAllocationException ex = new ResourceAllocationException("Cannot perform this operation, " + "Zone is currently disabled" + "zoneId=" + zone.getUuid(),
-                    ResourceType.network);
+            ResourceAllocationException ex = new ResourceAllocationException(generateErrorMessageForOperationOnDisabledZone("allocate Pod IP addresses", zone), ResourceType.network);
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/network/IpAddressManagerImpl.java:1138]: very long line :(

**Review by DaanHoogland — APPROVED** [review]

File: server/src/main/java/com/cloud/network/IpAddressManagerImpl.java (hunk 2 above)
Comment (GabrielBrascher) [reply server/src/main/java/com/cloud/network/IpAddressManagerImpl.java]: Good observation; updated code breaking line.

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Review by shwstppr — APPROVED** [review]
LGTM

**Comment by DaanHoogland** [comment]
@GabrielBrascher I have no opinion to speak of ;

> Additionally, method `com.cloud.network.NetworkModelImpl.areServicesEnabledInZone` does not use list `checkedProvider`; the respective list should either be removed or populated.
> 
> I have chosen to add provider names at `checkedProvider`, instead of removing it; however, I would appreciate any opinion/suggestion.

However, can you provide insight into your testing, i'm trusting the smoke tests a bit only. Any other testing needed?

**Comment by GabrielBrascher** [comment]
> @GabrielBrascher I have no opinion to speak of ;
> 
> > Additionally, method `com.cloud.network.NetworkModelImpl.areServicesEnabledInZone` does not use list `checkedProvider`; the respective list should either be removed or populated.
> > I have chosen to add provider names at `checkedProvider`, instead of removing it; however, I would appreciate any opinion/suggestion.
> 
> However, can you provide insight into your testing, i'm trusting the smoke tests a bit only. Any other testing needed?

@DaanHoogland  the `checkedProvider` list is created to be used in order to avoid executing ` isProviderEnabledInPhysicalNetwork(physicalNtwkId, providerName)` multiple times for the same network provider.

However, as it is not populated, the conditional `!checkedProvider.contains(providerName)` will always be true. Therefore that `List` is not usefull and loses its purpose. 

For reference, the code:
```
        List<String> checkedProvider = new ArrayList<String>();
        for (Service service : services) {
            // get all the providers, and check if each provider is enabled
            List<String> providerNames = _ntwkOfferingSrvcDao.listProvidersForServiceForNetworkOffering(offering.getId(), service);
            for (String providerName : providerNames) {
                if (!checkedProvider.contains(providerName)) {
                    result = result && isProviderEnabledInPhysicalNetwork(physicalNtwkId, providerName);
                }
                checkedProvider.add(providerName);
            }
```

**Pull request merged**